From 74e63b3acb0653d5aa6cb15edeb837f9bb371dac Mon Sep 17 00:00:00 2001
Message-Id: <patch-v4-3.7-74e63b3acb0-20210730T174308Z-avarab@gmail.com>
In-Reply-To: <cover-v4-0.7-00000000000-20210730T174308Z-avarab@gmail.com>
References: <cover-v3-0.7-00000000000-20210730T092843Z-avarab@gmail.com>
	<cover-v4-0.7-00000000000-20210730T174308Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 21 Jul 2021 23:36:28 +0200
Subject: [PATCH v4 3/7] http: drop support for curl < 7.17.0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In the last commit we dropped support for curl < 7.16.0, let's
continue that and drop support for versions older than 7.17.0. This
allows us to simplify the code by getting rid of some "#ifdef"'s to do
with the CURLOPT_USE_SSL flag.

The CURLOPT_USE_SSL flag was added in 7.17.0 as an alias for
CURLOPT_FTP_SSL[1].

The documentation[1] currently claims that it was introduced in
7.16.4, but the symbols-in-versions file correctly states
7.17.0[2].

I've submitted an upstream
patch (<patch-1.1-953bab490-20210730T170510Z-avarab@gmail.com>) to the
curl-library mailing list fix the documentation.

1. https://curl.se/libcurl/c/CURLOPT_USE_SSL.html
2. https://github.com/curl/curl/blob/master/docs/libcurl/symbols-in-versions

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 http.c | 2 --
 http.h | 9 ---------
 2 files changed, 11 deletions(-)

diff --git a/http.c b/http.c
index ef00e930232..c20f9b80303 100644
--- a/http.c
+++ b/http.c
@@ -933,10 +933,8 @@ static CURL *get_curl_handle(void)
 	if (curl_ftp_no_epsv)
 		curl_easy_setopt(result, CURLOPT_FTP_USE_EPSV, 0);
 
-#ifdef CURLOPT_USE_SSL
 	if (curl_ssl_try)
 		curl_easy_setopt(result, CURLOPT_USE_SSL, CURLUSESSL_TRY);
-#endif
 
 	/*
 	 * CURL also examines these variables as a fallback; but we need to query
diff --git a/http.h b/http.h
index cb092622a73..6758e5df347 100644
--- a/http.h
+++ b/http.h
@@ -16,15 +16,6 @@
 #define NO_CURL_EASY_DUPHANDLE
 #endif
 
-/*
- * CURLOPT_USE_SSL was known as CURLOPT_FTP_SSL up to 7.16.4,
- * and the constants were known as CURLFTPSSL_*
-*/
-#if !defined(CURLOPT_USE_SSL) && defined(CURLOPT_FTP_SSL)
-#define CURLOPT_USE_SSL CURLOPT_FTP_SSL
-#define CURLUSESSL_TRY CURLFTPSSL_TRY
-#endif
-
 struct slot_results {
 	CURLcode curl_result;
 	long http_code;
-- 
2.32.0.1071.g36f34456314

